src/main/java/com/svydovets/bibirnate/session/impl/JdbcEntityDao.java

```diff
@@ -0,0 +1,45 @@
+package com.svydovets.bibirnate.session.impl;
+
+import static com.svydovets.bibirnate.utils.EntityUtils.getFieldName;
+import static com.svydovets.bibirnate.utils.EntityUtils.getIdField;
+import static com.svydovets.bibirnate.utils.EntityUtils.getTableName;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+import javax.sql.DataSource;
+
+import com.svydovets.bibirnate.mapper.EntityMapperService;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+
+@RequiredArgsConstructor
+public class JdbcEntityDao {
+    public static final String SELECT_FROM_TABLE_BY_COLUMN = "select * from %s where %s = ?";
+    private final DataSource dataSource;
+
+    public <T> Optional<T> findById(Object id, Class<T> type) {
+        var field = getIdField(type);
+        return findBy(field, id, type);
+    }
+
+    @SneakyThrows
+    public <T> Optional<T> findBy(Field field, Object value, Class<T> type) {
+        try (var connection = dataSource.getConnection()) {
+            var tableName = getTableName(type);
+            var columnName = getFieldName(field);
+            String selectSql = String.format(SELECT_FROM_TABLE_BY_COLUMN, tableName, columnName);
```

Comment:
What if id is a String field? 
Should we take it into consideration and put it into single quotes?